src/framework/uicomponents/view/popupview.cpp

```diff
@@ -747,20 +747,22 @@ void PopupView::updateGeometry()
     bool canFitAbove = viewRect.height() < parentTopLeft.y();
     bool canFitBelow = viewRect.bottom() < anchorRect.bottom();
 
-    if (canFitAbove && (placement() == Above || !canFitBelow)) {
-        // place above the parent
-        qreal newY = parentTopLeft.y() - viewRect.height();
-        movePos(m_globalPos.x(), newY);
+    if (placement() == Placement::Above) {
+        movePos(m_globalPos.x(), parentTopLeft.y() - viewRect.height());
         setOpensUpward(true);

```

Comment:
What happens if there is no space at the top?

Reply:
Additional checks added